src/btree/bt_split.c

```diff
@@ -84,16 +85,19 @@ __split_verify_intl_key_order(WT_SESSION_IMPL *session, WT_PAGE *page)
 
         slot = 0;
         WT_INTL_FOREACH_BEGIN (session, page, ref) {
-            WT_ASSERT(session, ref->home == page);
+            WT_ASSERT_ALWAYS(
+              session, ref->home == page, "Failed to verify page reference after splitting");
 
             /*
              * Don't compare the first slot with any other slot, it's ignored on row-store internal
              * pages.
              */
             __wt_ref_key(page, ref, &next->data, &next->size);
             if (++slot > 2) {
-                WT_ASSERT(session, __wt_compare(session, btree->collator, last, next, &cmp) == 0);
-                WT_ASSERT(session, cmp < 0);
+                WT_ASSERT_ALWAYS(session,
+                  __wt_compare(session, btree->collator, last, next, &cmp) == 0,
```

Comment:
It is unusual but `__wt_compare` can return an error code, so what this is checking is that the call to the comparison function returns successfully as the subsequent check depends on it. A better message may be: "Key comparison failed during verification, cannot proceed."

Reply:
[4ec1860](https://github.com/wiredtiger/wiredtiger/pull/8623/commits/4ec1860657e6aeef19c7f7a20fa12817c7c791a1)